R/variables.R

```diff
@@ -104,65 +109,61 @@ create_variables <- function(pop, parameters) {
   create_age_variables(pop, parameters)
 }
 
-#' Adjust seeding ages
+#' @title Identify ages to adjust
 #'
-#' @details Switches age variables based on parameters object
+#' @details Identifies age variables to be switched based on parameters object
+#' @param discrete_ages discrete ages
 #' @param initial_values Vector of inital values from with variables object
 #'   created by \code{\link{create_variables}}
 #' @param parameters Parameters object created by \code{\link{get_parameters}}
-#' @return Returns modified initial_values vector
+#' @return Returns values to swap
 #' @importFrom utils head tail
 #'
 #' @noRd
-#' @examples
-#' \dontrun{
-#' Create our parameters
-#' pop <- squire::get_population(iso3c = "ATG")
-#' pop$n <- as.integer(pop$n)/100
-#' parameters <- get_parameters(
-#'    population = pop$n, contact_matrix_set = squire::contact_matrices[1]
-#' )
-#'
-#' # Create our variables
-#' variables <- create_variables(pop)
-#'
-#' # adjust the seeding ages
-#' variables$discrete_age$initial_values <- adjust_seeding_ages(
-#' initial_values = variables$discrete_age$initial_values,
-#'   parameters = parameters
-#' )
-#' }
-
-adjust_seeding_ages <- function(initial_values, parameters) {
+identify_ages_to_adjust <- function(discrete_ages, parameters) {
 
   # what are the ages that have been initialised
-  iv <- initial_values
+  iv <- discrete_ages
+
+  e1 <- parameters$E1_0
 
   # what ages need to be at the back of our initials for seeding
-  ages <- rep(which(parameters$E1_0 > 0),
-              parameters$E1_0[parameters$E1_0 > 0])
+  ages <- rep(which(e1 > 0), e1[e1 > 0])
 
-  # position of iv to be swapped out
+  # # position of iv to be swapped out
   to_distribute <- tail(seq_along(iv), length(ages))
 
   # position of iv to be swapped in
   to_swap <- vector()
+
   for (i in seq_along(unique(ages))) {
+    tsi1 <- list()
+    tsi2 <- list()
+    tsi1 <- which(iv == unique(ages)[i])
+    tsi2 <- head(tsi1, sum(ages == unique(ages)[i]))
```

Comment:
You might move `unique(ages)` up outside of this loop as it is constant and used twice. Longer term it should be possible to use something like `findInterval` to split this more efficiently but let's worry about that later if this flashes up in profiling